tests/network/expose.go

```diff
@@ -175,9 +226,20 @@ var _ = SIGDescribe("[rfe_id:253][crit:medium][vendor:cnv-qe@redhat.com][level:c
 				endpoint := endpoints.Subsets[0]
 				Expect(len(endpoint.Ports)).To(Equal(1))
 				Expect(endpoint.Ports[0].Port).To(Equal(int32(80)))
+
+				if ipFamily == ipv4 || ipFamily == ipv6 {
+					Expect(len(endpoints.Subsets[0].Addresses)).To(Equal(1))
+				} else {
+					assert.XFail(xfailError, func() {
+						Expect(len(endpoints.Subsets[0].Addresses)).To(Equal(2))
+					})
+				}
```

Comment:
To make this nicer, we can add a condition argument to `XFail`.
We are actually missing several useful arguments to control it, but it was not needed until now.

condition bool: Condition for marking the test function as xfail
With a condition, it will look like this:
```
expectedAddresses := 1
isDualStack := ipFamily == dualIpv4Primary || ipFamily == dualIpv4Primary
if isDualStack { expectedAddresses = 2 }
assert.XFail(isDualStack, xfailError, func() {...})
```

Reply:
done. thanks!